lib/shared/puppet_forge/v3/module_release.rb

```diff
@@ -0,0 +1,72 @@
+require 'shared/puppet_forge/connection'
+
+module PuppetForge
+  module V3
+    # Access metadata and downloads for a specific module release.
+    class ModuleRelease
+
+      include PuppetForge::Connection
+
+      # @!attribute [r] full_name
+      #   @return [Forge::ModuleName] The hyphen delimited name of this module
+      attr_reader :full_name
+
+      # @!attribute [r] version
+      #   @return [String] The version of this module
+      attr_reader :version
+
+      # @param full_name [String] The name of the module, will be converted to
+      #   a PuppetForge::ModuleName instance.
+      # @param version [String]
+      def initialize(full_name, version)
+        @full_name = PuppetForge::ModuleName.new(full_name)
+        @version   = version
+      end
+
+      # @return [Hash] The metadata for this module
+      def data
+        @data ||= conn.get(resource_url).body
+      end
```

Comment:
And in the longer term I intend to split out attr_readers or similar for each field - but for now I'm aiming for enough functionality to get things working without trying to build in more functionality than we need.